type/__package_apt/gencode-remote

```diff
@@ -60,11 +60,21 @@ aptget="DEBIAN_FRONTEND=noninteractive apt-get --quiet --yes -o Dpkg::Options::=
 case ${state_should}
 in
     present)
-        if test -n "${version_should}" -a "${version_should}" != "${version_is}"
+        if [ -f "$__object/parameter/source" ]
+        then
+            source_path="$(cat "$__object/parameter/source")"
+            tempfile="$($__remote_exec $__target_host "mktemp -t cidst.XXXXX.${source_path##*/}")"
+
+            $__remote_copy "$source_path" "$__target_host:$tempfile"
+
+            echo "${aptget:?} $recommendsparam install $tempfile"
+        elif test -n "${version_should}" -a "${version_should}" != "${version_is}"
         then
             # other version should be installed
             name="${name}=${version_should}"
             aptget="${aptget:?} -o APT::Get::allow-downgrades=true"
+
+            echo "${aptget} ${recommendsparam} install ${target_release} '${name}'"
```

Comment:
Moving the install command up here breaks the auto-update mechanism below (line 121ff.)

Reply:
Fair point. I shall have a think about this aspect.